source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
yes, that's the RFC, but I'm just explaining that it might not actually work. In fact, Envoy is broken in this regard. We don't correctly parse out port when doing virtual host / domain matching from the host/auth header. For the use case you are adding this for, I strongly suspect that you probably only need host and not port, but you know your use case.